bin/chef-cluster

```diff
@@ -0,0 +1,115 @@
+#!/bin/sh
+#
+# Copyright 2017, SUSE
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+cmd=$1
+workercount=${2:-auto}
+vhostconfig=/etc/apache2/vhosts.d/chef-cluster.conf
+serverconfig=/etc/sysconfig/chef-server
+webroot=/srv/www/chef-cluster
+
+baseport=4001
+
+balancerstart=BEGIN-BALANCER-MEMBER
+balancerend=END-BALANCER-MEMBER
+
+usage() {
+  echo "usage: chef-cluster <enable|disable> [<workercount>|auto]"
+  exit 1
+}
+
+case $cmd in
+  enable)
+    [ "$workercount" == "auto" ] && workercount=$(grep -c ^processor /proc/cpuinfo)
+    [ "$workercount" -ge 2 ] || { echo "ERROR: workercount must be >= 2"; exit 2; }
+
+    # apache configuration
+    mkdir -p "$webroot"
+    a2enmod -q lbmethod_byrequests || a2enmod lbmethod_byrequests
+    test -e "$vhostconfig" || cat <<EOF > $vhostconfig
+# configuration generated by: chef-cluster script
+Listen 4000
+
+<VirtualHost *:4000>
+  DocumentRoot /srv/www/chef-cluster
+  ErrorLog /var/log/apache2/chef-cluster-error.log
+  CustomLog /var/log/apache2/chef-cluster-access.log combined
+
+  <Location />
+    Require all granted
+  </Location>
+
+  <Proxy balancer://chefworkers>
+    # $balancerstart
+    # $balancerend
```

Comment:
Should we set some timeout/retry/failontimeout settings? See https://serverfault.com/a/609557 for instance.

Reply:
Ah, and also maybe explicitly set lbmethod here, in case several lbmethod modules are enabled?

Reply:
I was thinking about different methods but IMO in our case (i.e. when all workers run on the same host) there is very little difference between them.

Reply:
I will take a look at the timeouts.

Reply:
OK, I did some more checking:
1. `lbmethod=byrequests` is the default so even if some other ones are enabled, this will be used if nothing is selected explicitly
2. if I kill one of the workers, it's re-spawned but with different port (bug?). in such state failover works correctly out of the box without any special parameters.
3. if I start some dummy server on one of the ports (e.g. `netcat -l 4002`) to simulate hung worker the client times out after standard timeout (default 60sec). I can't find any option to make it failover to some other worker in such cases.

Reply:
RE 3. I added `failontimeout=On` which helps a bit by changing state of worker which timed out as 'error'. This state is kept for `retry` seconds (default 60).

Reply:
OK, I went through the code of `mod_proxy/mod_proxy_balancer` and it seems that failover after backend timeout is not implemented.